src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4713 @@
+                    dComIfGs_onEventBit(0x3201);
+                    mNextMessageNo = 0x164c;
+                }
+                else if (currMessageID == 0x164d) {
+                    mNextMessageNo = 0x164e;
+                }
+                else if (currMessageID == 0x164e) {
+                    mNextMessageNo = 0x164f;
+                }
+                else if (currMessageID == 0x164f) {
+                    dComIfGs_onEventBit(0x3380);
+                    if (dComIfGs_getTriforceNum() > 0) {
+                        mNextMessageNo = 0x1650;
+                    }
+                    else {
+                        mNextMessageNo = 0x1651;
+                    }
+                }
+                else if (currMessageID == 0x1683) {
+                    dComIfGs_onEventBit(0x3e20);
+                    if (dComIfGs_getItem(12) != dItem_NONE_e) {
+                        mNextMessageNo = 0x1685;
+                    }
+                    else {
+                        mNextMessageNo = 0x1684;
+                    }
+                }
+                return FALSE;
+            }
+        }
+    }
+    return TRUE;
 }
 
 /* 00001BD0-00001C24       .text seStart__8daShip_cFUlP4cXyz */
-void daShip_c::seStart(unsigned long, cXyz*) {
-    /* Nonmatching */
+void daShip_c::seStart(unsigned long i_seNum, cXyz* i_sePos) {
+    mDoAud_seStart(i_seNum, i_sePos, 0, m034A);
+    return;
 }
 
 /* 00001C24-00001D50       .text setSailAngle__8daShip_cFv */
 void daShip_c::setSailAngle() {
-    /* Nonmatching */
+    short sVar1;
+    cXyz* windVec = dKyw_get_wind_vec();
+    if (mTornadoActor) {
+        sVar1 = (short)(m03AA + 0x4000) - shape_angle.y;
+    }
+    else {
+        sVar1 = cM_atan2s(windVec->x, windVec->z) - shape_angle.y;
+    }
+    if (sVar1 > 0x800) {
+        m0380 = -0x1555;
+    }
+    else {
+        if (sVar1 < -0x800) {
+            m0380 = 0x1555;
+        }
+        else if (m0380 == 0) {
+            if (sVar1 >= 0) {
+                m0380 = -0x1555;
+            }
+            else {
+                m0380 = 0x1555;
+            }
+        }
+    }
+    cLib_addCalcAngleS(&m0364, m0380, 0x10, 0x800, 0x20);
+    m0378 = m0364 >> 1;
+    if (m0364 * m0380 < 0) {
+        if (checkStateFlg(daSFLG_UNK2_e) == 0) {
+            seStart(JA_SE_SHIP_SAIL_MOVE, &m0444);
+            onStateFlg(daSFLG_UNK2_e);
+        }
+    }
+    else {
+        offStateFlg(daSFLG_UNK2_e);
+    }
+    return;
 }
 
 /* 00001D50-00001E08       .text setControllAngle__8daShip_cFs */
-void daShip_c::setControllAngle(short) {
-    /* Nonmatching */
+void daShip_c::setControllAngle(short angle) {
+    short sVar1 = angle - m036C;
+    short sVar2 = sVar1 * 0.05f;
+    if (sVar2 == 0) {
+        if (sVar1 > 0) {
+            sVar2 = 1;
+        }
+        else if (sVar1 < 0) {
+            sVar2 = -1;
+        }
+    }
+    m036E += sVar2;
+    m036C += m036E;
+    cLib_addCalcAngleS(&m036E, 0, 0x14, 0x1000, 4);
+    return;
```

Comment:
`return;` at the end of all these void functions isn't needed, only in the middle of a function for an early return